core/templates/dev/head/services/AssetsBackendApiService.ts

```diff
@@ -79,7 +81,8 @@ angular.module('oppia').factory('AssetsBackendApiService', [
     // Map from asset filename to asset blob.
     var assetsCache = {};
```

Comment:
I see generic terms trough out this service (entityId, entityType) except line 72, 74 & 76, is that expected?

Reply:
Yeah, because those are lines related to uploading audio and this PR is only related to generalizing image upload in the frontend. 
In the backend, audio and images are stored in a similar fashion, but since only explorations have audio in it for now, in the frontend, the audio related functions are restricted to using exploration IDs.